### PR TITLE
add pool refresh interval to key value service config

### PR DESCRIPTION
**Pull request by jnelson15**

### CODE REVIEW

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraKeyValueServiceConfig.java [hunk 1]
```diff
@@ -43,6 +43,11 @@ public int poolSize() {
         return 20;
     }
 
+    @Value.Default
+    public int poolRefreshIntervalSeconds() {
+        return 5 * 60;
```
Comment (schlosna) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraKeyValueServiceConfig.java]: I don't have strong feelings about dropping down from 10 to 5 minute default, but we should document it... Somewhere

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java [hunk 2]
```diff
@@ -118,7 +118,7 @@ public void run() {
                     log.error("Failed to refresh Cassandra KVS pool. Extended periods of being unable to refresh will cause perf degradation.", t);
                 }
             }
-        }, CassandraConstants.SECONDS_BETWEEN_GETTING_HOST_LIST, CassandraConstants.SECONDS_BETWEEN_GETTING_HOST_LIST, TimeUnit.SECONDS);
+        }, config.poolRefreshIntervalSeconds(), config.poolRefreshIntervalSeconds(), TimeUnit.SECONDS);
```
Comment (schlosna) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java]: Not sure we should call this `poolRefreshIntervalSeconds()` as it seems a bit misleading -- it makes me think we're repopulating the backing connection pool every N seconds (e.g. maxLifetime In Hikari https://github.com/brettwooldridge/HikariCP#configuration-knobs-baby )

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraKeyValueServiceConfig.java (hunk 1 above)
Comment (jnelson15) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraKeyValueServiceConfig.java]: Do we track the values of timeouts/etc in CassandraConstants? We can set it back to 10 if people are potentially relying on this behavior. Mainly just wanted it configurable.

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java (hunk 2 above)
Comment (clockfort) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java]: maybe poolCheckIntervalSeconds ?

**Comment by clockfort** [comment]
lgtm.
You probably also want blacklist timeout to be configurable.

**Pull request merged**